### ISSUE TITLE
Fourier Transforms in X and Y

### ISSUE THREAD

**Issue by bendudson** [boutproject/BOUT-dev#21]
For flux-tube simulations, or turbulence simulations in periodic boxes, it is useful to be able to take Fourier transforms in directions other than Z: In Mesh there is currently the "Switch_YZ" function, which exists so that FFTs can be performed in Y. 

A more general method / API should be implemented, allowing FFTs in all dimensions. For some grids this may not make sense, in which cases exceptions should be thrown.

**Comment by d7919** [comment]
The switch_yz function has been removed now.

The rfft/irfft routines just takes pointers to the start of the input/output data so I think it should be possible to perform y based ffts by copying the data e.g. into a vector.

**Comment by dschwoerer** [comment]
Wouldn't be an mpi aware fftw easier/more scalable ... ?
http://www.fftw.org/fftw2_doc/fftw_4.html#SEC55

**Comment by d7919** [comment]
Yes in general it may well be better performing (although it depends what you're doing with the result I suppose). I was just noting that as a workaround with existing code if an fft in Y is needed it can be done with the existing routines.
